hack/configure-kcp.sh

```diff
@@ -21,48 +21,44 @@ function extra_params() {
-fi
-
-echo "Creating and accessing '${COMPUTE_WORKSPACE}' for compute:"
-KUBECONFIG=${KCP_KUBECONFIG} kubectl ws create ${COMPUTE_WORKSPACE} --type root:universal --ignore-existing || true
-KUBECONFIG=${KCP_KUBECONFIG} kubectl ws ${COMPUTE_WORKSPACE}
-echo
-
-SYNC_TARGET=appstudio-internal
-if [[ -z "$(kubectl get synctargets.workload.kcp.dev ${SYNC_TARGET} --kubeconfig ${KCP_KUBECONFIG} 2>/dev/null)" ]]; then
-  echo "Creating SyncTarget..."
-  KUBECONFIG=${KCP_KUBECONFIG} kubectl kcp workload sync ${SYNC_TARGET} --syncer-image ghcr.io/kcp-dev/kcp/syncer:main --resources=services,routes.route.openshift.io -o /tmp/${SYNC_TARGET}-syncer.yaml
-  if grep -q "insecure-skip-tls-verify: true" ${KCP_KUBECONFIG}; then
-    sed -i 's/certificate-authority-data: .*/insecure-skip-tls-verify: true/' /tmp/${SYNC_TARGET}-syncer.yaml
+function configure_compute_workspace() { 
+  # improve kubeconfig thing when this is addressed https://github.com/kcp-dev/kcp/issues/1689
+  echo "Using the '${ROOT_WORKSPACE}' workspace as the root"
+  KUBECONFIG=${KCP_KUBECONFIG} kubectl ws ${ROOT_WORKSPACE}
+  echo
+  
+  if [[ ${ROOT_WORKSPACE} == "root" ]]
+  then
+    COMPUTE_WORKSPACE=${COMPUTE_WORKSPACE:-"redhat-appstudio-internal-compute"}
+  else
+    COMPUTE_WORKSPACE=${COMPUTE_WORKSPACE:-"compute"}
   fi
-  kubectl apply -f /tmp/${SYNC_TARGET}-syncer.yaml --kubeconfig ${CLUSTER_KUBECONFIG}
-fi
-
-BIND_SCOPE="system:authenticated"
-if [[ ${KCP_INSTANCE_NAME} == "kcp-stable" ]] || [[ ${KCP_INSTANCE_NAME} == "kcp-unstable" ]] && [[ ${ROOT_WORKSPACE} == "root" ]]
-then
-  BIND_SCOPE="rh-sso:16270929"
-fi
-
-echo "Creating ClusterRole(Binding) to make the APIExport of the compute bindable for the group '${BIND_SCOPE}':"
-cat <<EOF | kubectl apply --kubeconfig ${KCP_KUBECONFIG} -f -
+  
+  echo "Creating and accessing '${COMPUTE_WORKSPACE}' for compute:"
+  KUBECONFIG=${KCP_KUBECONFIG} kubectl ws create ${COMPUTE_WORKSPACE} --type root:universal --ignore-existing || true
+  KUBECONFIG=${KCP_KUBECONFIG} kubectl ws ${COMPUTE_WORKSPACE}
+  echo
+  
+  SYNC_TARGET=appstudio-internal
+  if [[ -z "$(kubectl get synctargets.workload.kcp.dev ${SYNC_TARGET} --kubeconfig ${KCP_KUBECONFIG} 2>/dev/null)" ]]; then
+    echo "Creating SyncTarget..."
+    KUBECONFIG=${KCP_KUBECONFIG} kubectl kcp workload sync ${SYNC_TARGET} --syncer-image ghcr.io/kcp-dev/kcp/syncer:main --resources=services,routes.route.openshift.io -o /tmp/${SYNC_TARGET}-syncer.yaml
+    if grep -q "insecure-skip-tls-verify: true" ${KCP_KUBECONFIG}; then
+      sed -i 's/certificate-authority-data: .*/insecure-skip-tls-verify: true/' /tmp/${SYNC_TARGET}-syncer.yaml
+    fi
+    kubectl apply -f /tmp/${SYNC_TARGET}-syncer.yaml --kubeconfig ${CLUSTER_KUBECONFIG}
+  fi
+  
+  BIND_SCOPE="system:authenticated"
+  if [[ ${KCP_INSTANCE_NAME} == "kcp-stable" ]] || [[ ${KCP_INSTANCE_NAME} == "kcp-unstable" ]] && [[ ${ROOT_WORKSPACE} == "root" ]]
+  then
+    BIND_SCOPE="rh-sso:16270929"
```

Comment:
can we leave a comment here as to what this refers to?

Reply:
actually, I now see this was already here...nonetheless if you know, a comment would be nice

Reply:
added comment in 966caac
I hope that it clarifies the purpose of it